### DISCUSSION TITLE
Source code

### DISCUSSION THREAD

**Discussion by wchen342** [ungoogled-software/ungoogled-chromium-android#53] [General]
@csagan5 I recently discovered [https://github.com/macchrome/droidchrome/issues/5](https://github.com/macchrome/droidchrome/issues/5). Is this ever addressed?

**Comment by csagan5** [comment]
No, it is not, and I contacted the owner of https://chromium.woolyss.com/ around September last year asking at least to mention that there is **no source code** for those binary builds.

**Reply by wchen342** [reply]
I see. I checked their release page and it seems they stopped using bromite patches after that, at least they stopped mentioning it. Maybe they will pull ungoogled android build too since I switched license.

**Reply by wchen342** [reply]
Speaking too fast..it's still mentioned on their website. Just not in the Github release page. Huh.

**Reply by csagan5** [reply]
URL?

**Reply by wchen342** [reply]
https://chromium.woolyss.com/, scroll down to Android section and click on the line says "87.0.4280.101 (812852)".

**Reply by csagan5** [reply]
You're right, it is still linked from there.

**Reply by wchen342** [reply]
What do you do?

**Reply by csagan5** [reply]
There is not much that I can do; I expect the woolyss admin (Jerry) to do something eventually, I contacted him again.

I see that you changed the license to GPLv3 as well; what was the reason, if I may ask?

**Reply by csagan5** [reply]
Note: the biggest problem I see is not that the GPL license is violated there (which it is), but that people download binaries from random other people, without any bit of source available...

These same people are often contacted (can be daily) about injecting malware in their binaries.

**Reply by wchen342** [reply]
> what was the reason, if I may ask?

Well I cannot say it loudly right now but some company is interested in the patches and I think I'd better switch license to keep them available to all.

**Reply by wchen342** [reply]
> that people download binaries from random other people

Maybe you can put a warning on your site, although those who knows your site won't download it elsewhere I suppose.

**Reply by csagan5** [reply]
> some company is interested in the patches and I think I'd better switch license to keep them available to all

Interesting; it could be the same company that was interested in Kiwi's patches

**Reply by wchen342** [reply]
Actually I didn't know that story. Do you mind telling me a little more?

**Reply by csagan5** [reply]
@kiwibrowser can tell (if he wants or is allowed to)

**Reply by kiwibrowser** [reply]
@csagan5 At some point there was lot of interest from other developers and users about extensions, but I think this is going down now.
I understood now most users just care about adblocker.

When I was picking the license for Kiwi, I wasn't sure, and thought long time about the GPL case or not, what is the impact to users, to the project, to myself, etc.

I decided to go with 3-BSD clause for many reasons, one of them is that it gives lot of freedom (to everyone) and that enforcement of GPL is tricky, especially because the GPL rights cannot be secured properly in my opinion if the project is not GPL from inception A to Z.
Also it's the safest license, since it's used by Google itself.

There are some legal problems about how much anyone can rebrand MIT/BSD code to GPL (I concluded it was impossible); this is like "vi vs emacs", it's not a new problem, it exists since the *BSD and Linux like licenses fights.

For example, for ungoogled-chromium-android used some code from Kiwi for extensions, in theory you cannot relicense the derived work to GPL.
On the other side, ungoogled-chromium upstream / Bromite is GPL, so you couldn't license to BSD-3.

In practice, there is absolutely no reason to complain / spend time about it (for all the sides), because everybody benefits.

I guess this is the central piece of the question to ask yourself; what are you looking for ?

About @macchrome, I can agree with @csagan5 it's more a question of trusted builds / reproducible builds than the source-code itself because I don't think he did any additions.
At least if he had pointed to the source-repositories, I guess that would have been ok.

That would have been sort-of fair.

Just closing the issue and not answering is very rude. Like Davie504 would say: "NOT EPIC"...

Some companies anyway would find a way to bend the rules.
This reminds me the time when I asked the legal + open-source team @Microsoft for the source-code of Microsoft Edge for Android about 2 years ago (I think I even showed to @csagan5 at that time).

The MS legal approved my request, but the open-source team sent me back a bunch of .o files with a .sh file that was stitching the .o files together using the ld linker.
(the "source-code" is on the MS website now for anyone to enjoy!)

Hilarious and sad.

In my own perspective, I genuinely prefer, at least for the hundreds of hours that I spent working (on our respective projects), that the code is reused wildly / without contributions back than being too restrictive.

For now I'm getting more and more requests of developer support (which I decline) from Chinese Android browsers, so I guess clones are in the pipeline, maybe this is what is worrying you ?

**Reply by wchen342** [reply]
> in theory you cannot relicense the derived work to GPL.

I am not sure I get this part. I checked Kiwi's license and it is a slightly modified BSD 3-clause which is a GPL-compatible license so there shall be no problem? Actually the only files I used from Kiwi are the files related to `aura` so if necessary I can swap those out, but as long as you are not interested in starting a legal war I think we can put this aside. You did remind me that I forgot to put BSD notice in those files though. I shall add those back in next version.

EDIT: I see the problem, so the files licensed under BSD is not enforcable under GPL. It is not really a huge problem in this case though since there are only a few files.

> I genuinely prefer, at least for the hundreds of hours that I spent working (on our respective projects), that the code is reused wildly / without contributions back than being too restrictive.

I used to think that way too, but in reality I saw that companies has almost always being forcing things users don't want on them, and they simply don't care because users are always the vulnarable party. If they take what others shared and redistribute it or improve it that is of course welcomed, but unfortunately that is not happening in most cases. Companies are organizations and as an organization they have their agendas. I am a user myself, and I started this project exactly because I do not have a choice. Firefox has been slow and buggy (they did improve a lot in the past few year but they recently made mistakes again), and chromium send your data to Google (and there is no extensions). Other browsers like Icecat cannot make all the websites I need to use open. There has to be someone who does something for those who do not want to take those commercial "things", and using the license is one of the few things I can use to protect this freedom.

> For now I'm getting more and more requests of developer support (which I decline) from Chinese Android browsers, so I guess clones are in the pipeline, maybe this is what is worrying you ?

I can say for sure those clones will only be widely used inside China. I don't think they can get out, otherwise MS and Google will get to them first, no need for us to worry. The company I mentioned is not a Chinese one but they do have a well known chromium fork.

**Reply by kiwibrowser** [reply]
Nooooo :) If you need anything, feel free to take without worries; I feel from the code that you are like Expert level 999+ 🥇 on Extensions so no-one has to teach you anything on that.

I was just thinking about the topic and sharing open thoughts because from the latest commits, the licensing issue seemed to be in your mind currently, like if something upcoming was worrying you.

I was mentioning in general that it's tricky to enforce with lawyers because of this ambiguity between licenses.

By the way, about Chromium forks, I'm sure it can be interesting for you to talk to Vivaldi if one day you want an ethically-interesting job in Europe and still work on your code. I'm not sure if I ever mentioned, but this came into my mind again.

**Reply by wchen342** [reply]
Maybe it's just me being sensible to these topics because of the environment I was growing up :) I think the most important thing you taught me is that adding extensions to the android version is possibe, because I was hesitated to try this myself afraiding it's just a waste of time.

About the license problem, I think you are right about the hardness on enforcing them. But I think it is better to at least have something, even if it is just as good as a statement, than just sit and do nothing.

The thing I learned from recent experience that I'd better split what I am interested in and my job, that way I don't need to worry about both too much. Always prefer being more free.

**Comment by dm17** [comment]
How did this go for you guys? I wonder if you could use vanadium instead - since it is already setup for GrapheneOS, which should be similar enough to Android.

I wonder why there need to be so many independent efforts with a lot of the same goals (privacy & security)? I'm speaking, of course, about: 
1) ungoogled-chromium (this repo) 
2) ungoogled-chromium (Marmaduke's builds) 
3) Vanadium
4) Bromite 
5) potentially more I'm missing...

I realize that the maintainers goals don't line up entirely, but I'm surprised we can't find a way to deduplicate workloads & benefit more from each other's work.

EDIT: Reading this thread now, so I'll probably move my comment: https://github.com/Eloston/ungoogled-chromium/discussions/1395
